### PR TITLE
Get rid of numpy

### LINKED ISSUE #483: Eliminate dependency on numpy
numpy is large and we only use it a small number of places, which could fairly simply be replaced

### PR DESCRIPTION
**Pull request by cgccuser**
Closes #483

### CODE REVIEW

**Review by Lyxal — APPROVED** [review]

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -4349,7 +4349,7 @@
     lst: Diagonals of a matrix, starting with the main diagonal.
   vectorise: false
   tests:
-    - "[[[1,2,3],[4,5,6],[7,8,9]]] : [[1,5,9],[2,6],[3],[4,8],[7]]"
```
Comment (chunkybanana) [inline documents/knowledge/elements.yaml:4352]: Is this just a minor bugfix?

File: vyxal/elements.py [hunk 2]
```diff
@@ -135,26 +134,15 @@ def all_diagonals(lhs, ctx):
...
-            yield vyxalify(current_diagonal)
-            diag_num -= 1
-            current_diagonal = numpy.diag(vector, k=diag_num)
-
-    return gen()
+    vector = [iterable(x, ctx=ctx) for x in lhs]
+    all_diags = [[] for _ in range(len(vector) * 2 - 1)]
+    start = 0
+    print(vector)
+    for row in vector:
```
Comment (chunkybanana) [inline vyxal/elements.py:141]: This code looks like it relies on the matrix to be square...

File: vyxal/elements.py [hunk 3]
```diff
@@ -4058,11 +4049,7 @@ def vy_type(item, rhs=None, other=None, simple=False):
         )
     elif rhs is not None:
         return (vy_type(item, simple=simple), vy_type(rhs, simple=simple))
-    elif (
-        (x := type(item)) in (int, complex, float)
-        or is_sympy(item)
-        or isinstance(item, numpy.number)
-    ):
+    elif (x := type(item)) in (int, complex, float) or is_sympy(item):
```
Comment (chunkybanana) [inline vyxal/elements.py:4052]: `type` not `vy_type`?

File: vyxal/helpers.py [hunk 4]
```diff
@@ -487,11 +486,14 @@ def pad_to_square(array: VyList) -> VyList:
     Returns an array padded to the square of the largest dimension.
     https://stackoverflow.com/a/11763827/9363594
```
Comment (chunkybanana) [inline vyxal/helpers.py:487]: Probably get rid of this since it's not using that answer anymore

**Review by chunkybanana — COMMENTED** [review]
Just in general make sure there's nowhere that still uses numpy

File: documents/knowledge/elements.yaml (hunk 1 above)
Comment (cgccuser) [reply documents/knowledge/elements.yaml:4352]: Nah, it's just that my implementation returns them in a different order. We currently have no way for our tests to check things without the order, I should make an issue for that.

File: vyxal/elements.py (hunk 3 above)
Comment (cgccuser) [reply vyxal/elements.py:4052]: I think because `vy_type` returns Vyxal-specific types? @Lyxal wrote this, not me

File: vyxal/elements.py (hunk 2 above)
Comment (cgccuser) [reply vyxal/elements.py:141]: It does, why?

Comment (chunkybanana) [reply vyxal/elements.py:141]: If that's the spec, never mind.

File: vyxal/elements.py (hunk 3 above)
Comment (Lyxal) [reply vyxal/elements.py:4052]: Y'all realise that this is _inside_ the `vy_type` function right?

**Review by chunkybanana — APPROVED** [review]

**Pull request merged**